starknet/net/client.py

```diff
@@ -182,15 +182,15 @@ async def wait_for_tx(
 
             if status == TxStatus.ACCEPTED_ONCHAIN:
                 return result["block_number"], status
-            elif status == TxStatus.PENDING:
+            if status == TxStatus.PENDING:
                 if not wait_for_accept and "block_number" in result:
                     return result["block_number"], status
-            elif status == TxStatus.REJECTED:
+            if status == TxStatus.REJECTED:
                 raise Exception(f"Transaction [{tx_hash}] was rejected.")
-            elif status == TxStatus.NOT_RECEIVED:
+            if status == TxStatus.NOT_RECEIVED:
                 if not first_run:
                     raise Exception(f"Transaction [{tx_hash}] was not received.")
-            elif status != TxStatus.RECEIVED:
+            if status != TxStatus.RECEIVED:
                 raise Exception(f"Unknown status [{status}]")
```

Comment:
When `first_run = True` and `status  = TxStatus.NOT_RECEIVED` we'll throw `Unkown status`.